src/overlays/actors/ovl_En_Sth/z_en_sth.c

```diff
@@ -40,45 +48,357 @@ static ColliderCylinderInit D_80B0B404 = {
     },
     { 30, 40, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sth/func_80B07540.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sth/EnSth_Init.s")
+static s16 sObjectIds[6] = {
+    OBJECT_AHG, OBJECT_BOJ, OBJECT_BOJ, OBJECT_BOJ, OBJECT_BOJ, OBJECT_BOJ,
+};
+
+static FlexSkeletonHeader* sSkeletons[6] = {
+    0x060000F0, 0x060000F0, 0x060000F0, 0x060000F0, 0x060000F0, 0x060000F0,
+};
+
+static AnimationHeader* sAnimations[6] = {
+    &gParentDanceAnim, &gChildDanceAnim, &gChildDanceAnim, &gChildDanceAnim, &gChildDanceAnim, &gChildDanceAnim,
+};
+
+static EnSthActionFunc sRewardObtainedWaitActions[6] = {
+    EnSth_ParentRewardObtainedWait, EnSth_ChildRewardObtainedWait, EnSth_ChildRewardObtainedWait,
+    EnSth_ChildRewardObtainedWait,  EnSth_ChildRewardObtainedWait, EnSth_ChildRewardObtainedWait,
+};
+
+static u16 sEventFlags[6] = {
+    0x0000, 0x0400, 0x0800, 0x1000, 0x2000, 0x4000,
+};
+
+static s16 sGetItemIds[6] = {
+    GI_RUPEE_GOLD, GI_WALLET_ADULT, GI_STONE_OF_AGONY, GI_WALLET_GIANT, GI_BOMBCHUS_10, GI_HEART_PIECE,
+};
+
+static Vec3f D_80B0B49C = { 700.0f, 400.0f, 0.0f };
+
+static Color_RGB8 sTunicColors[6] = {
+    { 190, 110, 0 }, { 0, 180, 110 }, { 0, 255, 80 }, { 255, 160, 60 }, { 190, 230, 250 }, { 240, 230, 120 },
+};
+
+void EnSth_SetupAction(EnSth* this, EnSthActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnSth_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnSth* this = THIS;
+
+    s16 objectId;
+    s32 params = this->actor.params;
+    s32 objectBankIdx;
+    s32 params2;
+
+    // Translation: Gold Skulltula Shop
+    osSyncPrintf("\x1b[34m金スタル屋 no = %d\n\x1b[m", params);
```

Comment:
```suggestion
    osSyncPrintf(VT_FGCOL(BLUE) "金スタル屋 no = %d\n" VT_RST, params);
```
Note that you can use `./tools/vt_fmt.py "string to format"` to format strings properly.